### PR TITLE
OCM-11405 | fix: A few bug fixes for registry config

### PR DESCRIPTION
**Pull request by chenz4027**
https://issues.redhat.com/browse/OCM-11413
https://issues.redhat.com/browse/OCM-11405
https://issues.redhat.com/browse/OCM-11402
https://issues.redhat.com/browse/OCM-11404

### CODE REVIEW

File: cmd/edit/cluster/cmd.go [hunk 1]
```diff
@@ -639,6 +639,11 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 	if clusterRegistryConfigArgs != nil {
+		prompt := "Enabling cluster registry will cause rollout accross all nodepools. Do you want to proceed?"
```
Comment (chenz4027) [inline cmd/edit/cluster/cmd.go:642]: Current prompt looks like following.

If params set:
```
rosa git:(OCM-11405) ✗ ./rosa edit cluster -c magchen-hcp --registry-config-platform-allowlist 2e0auar98h9ajmd6ndlmlqr5lsibkm4v

? Enabling cluster registry will cause rollout accross all nodepools. Do you want to proceed? Yes
I: Updated cluster 'magchen-hcp'
```

Interactive mode:
```
➜  rosa git:(OCM-11405) ✗ ./rosa edit cluster -c magchen-hcp -i                                                                  
I: Interactive mode enabled.
Any optional fields can be ignored and will not be updated.
? Private cluster, check this command's help for possible impacts: No
? Disable Workload monitoring: No
? Update cluster-wide proxy: No
? Update additional trust bundle: No
? Update additional allowed principals: No
? Enable audit log forwarding to AWS CloudWatch: No
? Allowed Registries: abc.com
? Insecure Registries: insecure.com
? Allowed Registries For Import: quay.io:true
? Registry Additional Trusted CA (optional): 
? Enabling cluster registry will cause rollout accross all nodepools. Do you want to proceed? Yes
? Enable cluster deletion protection: No
I: Updated cluster 'magchen-hcp'
```

Comment (chenz4027) [reply cmd/edit/cluster/cmd.go]: I miss spelled across to "accross", now it's fixed.

File: cmd/edit/cluster/cmd.go [hunk 2]
```diff
@@ -639,6 +639,11 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 	if clusterRegistryConfigArgs != nil {
+		prompt := "Enabling cluster registry will cause rollout across all nodepools. Do you want to proceed?"
```
[the remark this answers is not in the document]
Comment (chenz4027) [reply cmd/edit/cluster/cmd.go:642]: Good catch @andreadecorte !

Comment (arendej) [inline cmd/edit/cluster/cmd.go:642]: change text to:

`Warning: Enabling or editing the cluster registry configuration will cause a rollout across all machinepools (all machinepool nodes will be recreated, following pod draining from each node). Do you want to proceed?`

File: cmd/edit/cluster/cmd.go [hunk 3]
```diff
@@ -639,6 +639,11 @@ func run(cmd *cobra.Command, _ []string) {
...
+		if !confirm.ConfirmRaw(prompt) {
+			r.Reporter.Warnf("You opted out from enabling a cluster with registry config.")
```
Comment (arendej) [inline cmd/edit/cluster/cmd.go:644]: change text to:
`You have not changed any registry configuration -- exiting.`

File: pkg/clusterregistryconfig/flags.go [hunk 4]
```diff
@@ -152,25 +159,30 @@ func GetClusterRegistryConfigOptions(cmd *pflag.FlagSet,
...
-			return nil, fmt.Errorf("Expected a valid value for allowed registries: %s", err)
+		// Allowed registries and blocked registries are mutually exclusive
+		if result.blockedRegistries == nil {
+			allowedRegistriesInputs, err := interactive.GetString(interactive.Input{
+				Question: "Allowed Registries",
+				Help:     cmd.Lookup(allowedRegistriesFlag).Usage,
+				Default:  strings.Join(defaultAllowedRegistries, ","),
+			})
+			if err != nil {
+				return nil, fmt.Errorf("Expected a valid value for allowed registries: %s", err)
```
Comment (arendej) [inline pkg/clusterregistryconfig/flags.go:170]: what is a valid value? can we show an example or explain if it's a comma separated string?

File: pkg/clusterregistryconfig/flags.go [hunk 5]
```diff
@@ -152,25 +159,30 @@ func GetClusterRegistryConfigOptions(cmd *pflag.FlagSet,
...
-		if err != nil {
-			return nil, fmt.Errorf("Expected a valid value for blocked registries: %s", err)
+		if result.allowedRegistries == nil {
+			blockedRegistriesInputs, err := interactive.GetString(interactive.Input{
+				Question: "Blocked Registries",
+				Help:     cmd.Lookup(blockedRegistriesFlag).Usage,
+				Default:  strings.Join(defaultBlockedRegistries, ","),
+			})
+			if err != nil {
+				return nil, fmt.Errorf("Expected a valid value for blocked registries: %s", err)
```
Comment (arendej) [inline pkg/clusterregistryconfig/flags.go:182]: same comment as for allowed registries above

**Review by arendej — CHANGES_REQUESTED** [review]

**Comment by chenz4027** [comment]
Updated the text, I combined the suggestions from both @andreadecorte and @arendej . Would you mind taking another look?

File: cmd/edit/cluster/cmd.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (andreadecorte) [reply cmd/edit/cluster/cmd.go]:
```
./rosa edit cluster -c ad-loc7 --registry-config-insecure-registries insecure.io,insecure2.io,insecure3.io
? Changing any registry related parameter will trigger a rollout across all machinepools (all machinepool nodes will be recreated, following pod draining from each node). Do you want to proceed? (y/N)
W: You have not changed any registry configuration -- exiting.
```
looks good now

**Review by arendej — APPROVED** [review]
thanks for the changes

**Pull request merged**